src/commands/deployImage/imageSource/containerRegistry/acr/AcrListStep.ts

```diff
@@ -69,3 +112,22 @@ export class AcrListStep extends AzureWizardPromptStep<IContainerRegistryImageCo
     }
 }
 
+async function tryConfigureResourceGroupForRegistry(
+    context: IContainerRegistryImageContext,
+    promptSteps: AzureWizardPromptStep<IContainerRegistryImageContext>[],
+): Promise<void> {
+    // No need to pollute the base context with all the potential pre-create typings as they are not otherwise used
+    const resourceCreationContext = context as Partial<ICreateContainerAppContext> & Partial<IManagedEnvironmentContext> & CreateAcrContext;
+    if (resourceCreationContext.resourceGroup || resourceCreationContext.newResourceGroupName) {
+        return;
+    }
+
+    // Try to leverage an existing resource group (managed environment name is a fallback here because our create flow has the resource group name mirror the environment name)
```

Comment:
Shouldn't you check what resource group the managed environment is in rather than using its name?

Reply:
Yes, this would be the wiser way to do it.  My brain must not have been working properly when I wrote this part 😆